<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>UTBiocomputing2015</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-light.css">
    <script src="javascripts/scale.fix.js"></script>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1 class="header">UT Biocomputing 2015</h1>
        <p class="header">Peer-led working group in Biological Computing at UT Austin, Spring 2015</p>

        <ul>
          <li class="download"><a class="buttons" href="https://github.com/sjspielman/UTbiocomputing2015/zipball/master">Download ZIP</a></li>
          <li class="download"><a class="buttons" href="https://github.com/sjspielman/UTbiocomputing2015/tarball/master">Download TAR</a></li>
          <li><a class="buttons github" href="https://github.com/sjspielman/UTbiocomputing2015">View On GitHub</a></li>
        </ul>

        <p class="header">This project is maintained by <a class="header name" href="https://github.com/sjspielman">sjspielman</a></p>


      </header>
      <section>

<h2>
<a id="welcome" class="anchor" href="#welcome" aria-hidden="true"><span class="octicon octicon-link"></span></a>Welcome!</h2>

<p>
This is the official website for the Spring 2015 Peer-led Biocomputing Group at UT Austin, offered through the <a href="http://ccbb.biosci.utexas.edu/">Center for Computational Biology and Bioinformatics</a>. The goal of this class is to take away any fear of programming that you may have so that computing can become an advantage rather than a barrier to your research. On Wednesdays, we will provide lessons suitable for beginners in programming. Each week you should post at least one question and answer on the <a href="http://goo.gl/PLnSpP">UTbiocomputing google group</a> and complete the homework. On Tuesdays we will help you with homework or if you had no problems, come anyways to meet other people that are computing at UT, and get some handy tips from GSAF's Scott Hunicke-Smith.
<h5>Please install the required software no later than the second day of class. Installation instructions are available <a href="https://github.com/sjspielman/UTbiocomputing2015/blob/master/install.md">here</a>.</h5>


<h2> Instructors </h2>
<li>
<a href="http://www.rebeccatarvin.com">Rebecca (Becca) Tarvin</a> (PAT 123) rdtarvin@utexas.edu 
</li>
<li>
<a href="http://www.sjspielman.org">Stephanie Spielman</a> (MBB 3.232) stephanie.spielman@gmail.com
</li>

<h2> Meeting Times </h2>
<li>
Open Coding Hour: Tuesdays 5-6pm in CCBB conference room (GDC 7.514) 
</li>
<li>
Class: 4-5pm on Wednesdays in FNT 1.104
</li>
</p>

<h2> <a id="resources" class="anchor" href="#resources" aria-hidden="true"><span class="octicon octicon-link"></span></a>Helpful Resources</h2>
<p>

<li> 
We strongly recommend the book <a href="http://practicalcomputing.org">Practical Computing for Biologists</a>. by Haddock and Dunn. The book provides a really thorough overview of most of what we'll learn in this class (including Unix, Python, R, , and more! The book's accompanying website (linked above) is also regularly maintained with important tips, examples, and errata.
</li><br>

<li> 
The <a href="http://www.ee.surrey.ac.uk/Teaching/Unix/">UNIX Tutorial for Beginners</a> is a great resource and starting point for getting comfortable with the command-line environment.
</li><br>

<li>
Don't forget your most valuable resource: google! If you encounter an issue, chances are somebody else has also encountered it and has asked about it. Googling your error messages is one of the best debugging strategies there is. In particular, try to find links to the website [**stackoverflow.com**](http://www.stackoverflow.com). This forum-based website has all the answers, possibly literally.
</li><br>

<li>
<a href="https://wikis.utexas.edu/display/CCBB/Introduction+to+Biological+Computing+Course">This wiki</a> and <a href="https://github.com/wrightaprilm/BioComputing2014/tree/master">this github repository</a> contain powerpoints, exercises, and cheatsheets from the Spring 2014 course.
</li><br>

<li>
Here's a great paper on <a href="(http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001745">Best Practices in Scientific Computing</a>.
</li>

</p>

<h2> <a id="schedule" class="anchor" href="#schedule" aria-hidden="true"><span class="octicon octicon-link"></span></a>Schedule and Materials</h2>
<p>
Each item in the "Description" column links to materials for that course day. Materials include a cheatsheet, lesson plan, and sometimes a powerpoint and/or example scripts. All course materials can also be accessed directly from <a href="https://github.com/sjspielman/UTbiocomputing2015/"> this github repository</a>.
</p>

<table>
  <tr>
    <th>Week</th>
    <th>Date</th> 
    <th>Topic</th>
    <th>Description</th>
    <th>Instructor</th>
  </tr>
  <tr>
    <td>Week 1</td>
    <td>Jan 21</td> 
    <td>Introduction</td>
    <td>Welcome!</td>
    <td>Becca</td>
  </tr>

  <tr>
    <td>Week 2</td>
    <td>Jan 28</td> 
    <td>UNIX and Bash</td>
    <td>Navigating the Command Line</td>
    <td>Becca</td>
  </tr>

  <tr>
    <td>Week 3</td>
    <td>Feb 04</td> 
    <td>Python I</td>
    <td>Basic data types and structure</td>
    <td>Stephanie</td>
  </tr>

  <tr>
    <td>Week 4</td>
    <td>Feb 11</td> 
    <td>Python II</td>
    <td>Control flow and loops</td>
    <td>Becca</td>
  </tr>
  
    <tr>
    <td>Week 5</td>
    <td>Feb 18</td> 
    <td>Python III</td>
    <td>Functions</td>
    <td>Stephanie</td>
  </tr>
  
    <tr>
    <td>Week 6</td>
    <td>Feb 25</td> 
    <td>Python IV</td>
    <td>File Input/Output</td>
    <td>Stephanie</td>
  </tr>
  
    <tr>
    <td>Week 7</td>
    <td>Mar 04</td> 
    <td>Python V</td>
    <td>Testing and Code Hygiene</td>
    <td>Becca</td>
  </tr>
  
    <tr>
    <td>Week 8</td>
    <td>Mar 11</td> 
    <td>Python VI</td>
    <td>BioPython</td>
    <td>Stephanie</td>
  </tr>
  
    <tr>
    <td>Week 9</td>
    <td>Mar 25</td> 
    <td>"PyBash"</td>
    <td>Merging Python and Bash</td>
    <td>Stephanie and Becca</td>
  </tr>
  
    <tr>
    <td>Week 10</td>
    <td>Apr 01</td> 
    <td>Version Control</td>
    <td>git</td>
    <td>Cheng Lee</td>
  </tr>
  
    <tr>
    <td>Week 11</td>
    <td>Apr 08</td> 
    <td>Stastical Computing</td>
    <td>R</td>
    <td>Nate Pope</td>
  </tr>
  
    <tr>
    <td>Week 12</td>
    <td>Apr 15</td> 
    <td>Data Analysis in Python</td>
    <td>pandas</td>
    <td>Ben Liebeskind</td>
  </tr>

    <tr>
    <td>Week 13</td>
    <td>Apr 22</td> 
    <td>High-performance computing</td>
    <td>TACC</td>
    <td>Benni Goetz</td>
  </tr>

      <tr>
    <td>Week 14</td>
    <td>Apr 29</td> 
    <td>RNAseq Analysis</td>
    <td>Computational platforms and pipelines</td>
    <td>Dariya Sydykova</td>
  </tr>
  
      <tr>
    <td>Week 15</td>
    <td>May 06</td> 
    <td>pyRAD</td>
    <td>Pipelines for analyzing NextGen SNP data</td>
    <td>April Wright</td>
  </tr>

</table>




      </section>
      <footer>
        <p><small>Hosted on <a href="https://pages.github.com">GitHub Pages</a> using the Dinky theme, modified by sjspielman.</small></p>
      </footer>
    </div>
    <!--[if !IE]><script>fixScale(document);</script><![endif]-->
		
  </body>
</html>
